### PR TITLE
chore: unpin nodejs version from 22.4

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Use the latest stable node release. 

**Description**

Node version [22.5.1](https://github.com/nodejs/node/releases/tag/v22.5.1) fixes the regressions. 

**Steps to test or reproduce**

Run all tests.

### CODE REVIEW

**Review by nflaig — COMMENTED** [review]
Is there a good reason to update to 22.5 now? I would be in favor of keeping 22.4 until node 22 becomes the stable release.

Based on the diagram it should be in October which is not too far out
image

**Comment by wemeetagain** [comment]
moving to draft as it seems we have some consensus to wait until 22 becomes active LTS

File: .github/workflows/docs-check.yml [hunk 1]
```diff
@@ -15,8 +15,8 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: 22.4
-          cache: yarn
+          node-version: 22
+          cache: yarn
```
Comment (nflaig) [inline .github/workflows/docs-check.yml]: we shouldn't re-add the white space here, and in a few other places

Comment (nazarhussain) [reply .github/workflows/docs-check.yml]: Didn't added anything manually, just reverted the earlier merged PR.

**Comment by nazarhussain** [comment]
> moving to draft as it seems we have some consensus to wait until 22 becomes active LTS

@wemeetagain  Previously we didn't had precedence to only use LTS versions, rather latest version. Even current codebase is still locked upto `22.4`. So either we change it and completely strict 22 or use latest version of 22. 

This PR was reverting an earlier PR because 22.5 introduced a regression, but 22.5.1 fixed it. Don't see any reason to keep restricting upto 22.4.

File: .github/workflows/docs-check.yml (hunk 1 above)
Comment (nflaig) [reply .github/workflows/docs-check.yml]: yes, but if/when we merge this should clean up the diff before that

**Comment by nflaig** [comment]
> This PR was reverting an earlier PR because 22.5 introduced a regression, but 22.5.1 fixed it. Don't see any reason to keep restricting upto 22.4.

I personally don't have confidence in nodejs release process for `current`, apparently their CI had errors but still passed. The LTS release runs much more tests, does a whole range of ecosystem smoke tests, and is a whole less likely to break.

Having to debug strange node issue is not fun..and we have better things to do

**Comment by wemeetagain** [comment]
> Previously we didn't had precedence to only use LTS versions, rather latest version.

Yes, I think our experience the other day, where we had to pin the version, changed my mind a little bit.

It seems that the nodejs folks treat the new version with more leeway to be actually unstable and breaking before it becomes "current". Once it becomes "current", their standards for something breaking become much higher.

I tend to agree with Nico that we don't really care to have our daily progress possibly stopped again by a breaking version, and we can mitigate that by pinning a specific version until it becomes "current" LTS.

**Review by nflaig — APPROVED** [review]
Nodejs 22 is in LTS now https://nodejs.org/en/blog/release/v22.11.0

**Comment by nazarhussain** [comment]
@wemeetagain Need to update the required check names to align with node version.

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
rerunning CI so the settings will remember the new jobs to enforce (settings only remember CI job names run in the last week)

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.24.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.24.0) :tada:

**Pull request merged**